File: src/main/java/com/example/demo/controller/GeoIpDebugController.java
package com.example.demo.controller;

import com.example.demo.service.GeoIpCountryService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class GeoIpDebugController {

    private final GeoIpCountryService geoIpCountryService;

    public GeoIpDebugController(GeoIpCountryService geoIpCountryService) {
        this.geoIpCountryService = geoIpCountryService;
    }

    @GetMapping("/api/geoip/status")
    public GeoIpStatus status(@RequestParam(required = false) String ip) {
        boolean enabled = geoIpCountryService.isEnabled();
        String testIp = (ip != null && !ip.isBlank()) ? ip : "1.1.1.1";
        String cc = enabled ? geoIpCountryService.lookupCountryCode(testIp) : null;
        return new GeoIpStatus(enabled, testIp, cc);
    }

    public record GeoIpStatus(boolean enabled, String testIp, String countryCode) {
    }
}

File: src/main/kotlin/com/example/demo/controller/GeoIpDebugController.kt
package com.example.demo.controller

import com.example.demo.service.GeoIpCountryService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
class GeoIpDebugController(
    private val geoIpCountryService: GeoIpCountryService
) {
    data class GeoIpStatus(
        val enabled: Boolean,
        val testIp: String?,
        val countryCode: String?
    )

    @GetMapping("/api/geoip/status")
    fun status(@RequestParam(required = false) ip: String?): GeoIpStatus {
        val enabled = geoIpCountryService.isEnabled()
        val testIp = ip ?: "1.1.1.1"
        val cc = if (enabled) geoIpCountryService.lookupCountryCode(testIp) else null
        return GeoIpStatus(enabled = enabled, testIp = testIp, countryCode = cc)
    }
}

